### ISSUE TITLE
JSONDecodeError 

### ISSUE THREAD

**Issue by briandrawert** [StochSS/stochss-compute#47]
To replicate check out the 'docs/readme' branch.
```
============= in a terminal
python3 -m venv venv
source venv/bin/activate
python3 app.py

============= in a 2nd terminal
source venv/bin/activate
jupyter notebook --port=9999

============= in your web browser go to:
http://localhost:9999/notebooks/examples/StartHere.ipynb

========== Make the 2nd cell the following:
results = RemoteSimulation.on(ComputeServer(
    "127.0.0.1", port=1234)).with_model(model).run()
```

Gives the following error:

```
http://127.0.0.1:1234/api/v1/gillespy2/model/run
---------------------------------------------------------------------------
JSONDecodeError                           Traceback (most recent call last)
~/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/pydantic/main.cpython-38-x86_64-linux-gnu.so in pydantic.main.BaseModel.parse_raw()

~/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/pydantic/parse.cpython-38-x86_64-linux-gnu.so in pydantic.parse.load_str_bytes()

/usr/lib/python3.8/json/__init__.py in loads(s, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)
    356             parse_constant is None and object_pairs_hook is None and not kw):
--> 357         return _default_decoder.decode(s)
    358     if cls is None:

/usr/lib/python3.8/json/decoder.py in decode(self, s, _w)
    336         """
--> 337         obj, end = self.raw_decode(s, idx=_w(s, 0).end())
    338         end = _w(s, end).end()

/usr/lib/python3.8/json/decoder.py in raw_decode(self, s, idx)
    354         except StopIteration as err:
--> 355             raise JSONDecodeError("Expecting value", s, err.value) from None
    356         return obj, end

JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

ValidationError                           Traceback (most recent call last)
/tmp/ipykernel_676520/1788521989.py in <module>
      3 model2 = MichaelisMenten()
      4 
----> 5 results = RemoteSimulation.on(ComputeServer(
      6     "127.0.0.1", port=1234)).with_model(model).run()
      7 # results.plot()

~/Desktop/research/stochss-compute/stochss_compute/remote_simulation.py in run(self, **params)
     66 
     67         start_request = ModelRunRequest(model=self.model, kwargs=params)
---> 68         start_response = unwrap_or_err(JobStatusResponse, self.server.post(Endpoint.GILLESPY2_MODEL, sub="/run", request=start_request))
     69 
     70         remote_results = RemoteResults(result_id=start_response.job_id, server=self.server)

~/Desktop/research/stochss-compute/stochss_compute/remote_utils.py in unwrap_or_err(response_model, response)
      8 def unwrap_or_err(response_model: type[BaseModel], response: Response):
      9     if not response.ok:
---> 10         raise Exception(ErrorResponse.parse_raw(response.text).msg)
     11 
     12     return response_model.parse_raw(response.text)

~/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/pydantic/main.cpython-38-x86_64-linux-gnu.so in pydantic.main.BaseModel.parse_raw()

ValidationError: 1 validation error for ErrorResponse
__root__
  Expecting value: line 1 column 1 (char 0) (type=value_error.jsondecode; msg=Expecting value; doc=<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
  "http://www.w3.org/TR/html4/loose.dtd">
<html>
  <head>
    <title>AttributeError: &#x27;DaskDelegateConfig&#x27; object has no attribute &#x27;cluster_address&#x27; // Werkzeug Debugger</title>
    <link rel="stylesheet" href="?__debugger__=yes&amp;cmd=resource&amp;f=style.css"
        type="text/css">
    <!-- We need to make sure this has a favicon so that the debugger does
         not accidentally trigger a request to /favicon.ico which might
         change the application's state. -->
    <link rel="shortcut icon"
        href="?__debugger__=yes&amp;cmd=resource&amp;f=console.png">
    <script src="?__debugger__=yes&amp;cmd=resource&amp;f=debugger.js"></script>
    <script type="text/javascript">
      var TRACEBACK = 140061202724032,
          CONSOLE_MODE = false,
          EVALEX = true,
          EVALEX_TRUSTED = false,
          SECRET = "luXQi8xyK2LTzlZBO9G1";
    </script>
  </head>
  <body style="background-color: #fff">
    <div class="debugger">
<h1>AttributeError</h1>
<div class="detail">
  <p class="errormsg">AttributeError: &#x27;DaskDelegateConfig&#x27; object has no attribute &#x27;cluster_address&#x27;</p>
</div>
<h2 class="traceback">Traceback <em>(most recent call last)</em></h2>
<div class="traceback">
  <h3></h3>
  <ul><li><div class="frame" id="frame-140061202764896">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/stochss_compute/api/delegate/dask_delegate.py"</cite>,
      line <em class="line">57</em>,
      in <code class="function">__init__</code></h4>
  <div class="source "><pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>super()</pre>
<pre class="line before"><span class="ws">        </span>self.delegate_config = delegate_config</pre>
<pre class="line before"><span class="ws">        </span># # Attempt to load the global mDask client.</pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line current"><span class="ws">            </span>self.client = get_client()</pre>
<pre class="line after"><span class="ws">        </span>except ValueError as _:</pre>
<pre class="line after"><span class="ws">            </span>if self.delegate_config.kube_cluster is not None:</pre>
<pre class="line after"><span class="ws">                </span>self.client = Client(self.delegate_config.kube_cluster)</pre>
<pre class="line after"><span class="ws">                </span>print(self.delegate_config.kube_cluster)</pre>
<pre class="line after"><span class="ws">            </span># TODO what happens if user is not using kubernetes?</pre></div>
</div>

<li><div class="frame" id="frame-140061202765376">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/distributed/worker.py"</cite>,
      line <em class="line">4088</em>,
      in <code class="function">get_client</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">    </span>if client and (not address or client.scheduler.address == address):</pre>
<pre class="line before"><span class="ws">        </span>return client</pre>
<pre class="line before"><span class="ws">    </span>elif address:</pre>
<pre class="line before"><span class="ws">        </span>return Client(address, timeout=timeout)</pre>
<pre class="line before"><span class="ws">    </span>else:</pre>
<pre class="line current"><span class="ws">        </span>raise ValueError(&quot;No global client found and no address provided&quot;)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws"></span>def secede():</pre>
<pre class="line after"><span class="ws">    </span>&quot;&quot;&quot;</pre>
<pre class="line after"><span class="ws">    </span>Have this task secede from the worker&#x27;s thread pool</pre></div>
</div>

<li><div class="exc-divider">During handling of the above exception, another exception occurred:</div>
<li><div class="frame" id="frame-140061202724464">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py"</cite>,
      line <em class="line">2091</em>,
      in <code class="function">__call__</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">    </span>def __call__(self, environ: dict, start_response: t.Callable) -&gt; t.Any:</pre>
<pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;The WSGI server calls the Flask application object as the</pre>
<pre class="line before"><span class="ws">        </span>WSGI application. This calls :meth:`wsgi_app`, which can be</pre>
<pre class="line before"><span class="ws">        </span>wrapped to apply middleware.</pre>
<pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
<pre class="line current"><span class="ws">        </span>return self.wsgi_app(environ, start_response)</pre></div>
</div>

<li><div class="frame" id="frame-140061202724752">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py"</cite>,
      line <em class="line">2076</em>,
      in <code class="function">wsgi_app</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>try:</pre>
<pre class="line before"><span class="ws">                </span>ctx.push()</pre>
<pre class="line before"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
<pre class="line before"><span class="ws">            </span>except Exception as e:</pre>
<pre class="line before"><span class="ws">                </span>error = e</pre>
<pre class="line current"><span class="ws">                </span>response = self.handle_exception(e)</pre>
<pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
<pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre>
<pre class="line after"><span class="ws">                </span>raise</pre>
<pre class="line after"><span class="ws">            </span>return response(environ, start_response)</pre>
<pre class="line after"><span class="ws">        </span>finally:</pre></div>
</div>

<li><div class="frame" id="frame-140061202724656">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py"</cite>,
      line <em class="line">2073</em>,
      in <code class="function">wsgi_app</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span>ctx = self.request_context(environ)</pre>
<pre class="line before"><span class="ws">        </span>error: t.Optional[BaseException] = None</pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>try:</pre>
<pre class="line before"><span class="ws">                </span>ctx.push()</pre>
<pre class="line current"><span class="ws">                </span>response = self.full_dispatch_request()</pre>
<pre class="line after"><span class="ws">            </span>except Exception as e:</pre>
<pre class="line after"><span class="ws">                </span>error = e</pre>
<pre class="line after"><span class="ws">                </span>response = self.handle_exception(e)</pre>
<pre class="line after"><span class="ws">            </span>except:  # noqa: B001</pre>
<pre class="line after"><span class="ws">                </span>error = sys.exc_info()[1]</pre></div>
</div>

<li><div class="frame" id="frame-140061202724944">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py"</cite>,
      line <em class="line">1518</em>,
      in <code class="function">full_dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>request_started.send(self)</pre>
<pre class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
<pre class="line before"><span class="ws">            </span>if rv is None:</pre>
<pre class="line before"><span class="ws">                </span>rv = self.dispatch_request()</pre>
<pre class="line before"><span class="ws">        </span>except Exception as e:</pre>
<pre class="line current"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
<pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def finalize_request(</pre>
<pre class="line after"><span class="ws">        </span>self,</pre>
<pre class="line after"><span class="ws">        </span>rv: t.Union[ResponseReturnValue, HTTPException],</pre></div>
</div>

<li><div class="frame" id="frame-140061202725184">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py"</cite>,
      line <em class="line">1516</em>,
      in <code class="function">full_dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span>self.try_trigger_before_first_request_functions()</pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line before"><span class="ws">            </span>request_started.send(self)</pre>
<pre class="line before"><span class="ws">            </span>rv = self.preprocess_request()</pre>
<pre class="line before"><span class="ws">            </span>if rv is None:</pre>
<pre class="line current"><span class="ws">                </span>rv = self.dispatch_request()</pre>
<pre class="line after"><span class="ws">        </span>except Exception as e:</pre>
<pre class="line after"><span class="ws">            </span>rv = self.handle_user_exception(e)</pre>
<pre class="line after"><span class="ws">        </span>return self.finalize_request(rv)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def finalize_request(</pre></div>
</div>

<li><div class="frame" id="frame-140061202724320">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py"</cite>,
      line <em class="line">1502</em>,
      in <code class="function">dispatch_request</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">            </span>getattr(rule, &quot;provide_automatic_options&quot;, False)</pre>
<pre class="line before"><span class="ws">            </span>and req.method == &quot;OPTIONS&quot;</pre>
<pre class="line before"><span class="ws">        </span>):</pre>
<pre class="line before"><span class="ws">            </span>return self.make_default_options_response()</pre>
<pre class="line before"><span class="ws">        </span># otherwise dispatch to the handler for that endpoint</pre>
<pre class="line current"><span class="ws">        </span>return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def full_dispatch_request(self) -&gt; Response:</pre>
<pre class="line after"><span class="ws">        </span>&quot;&quot;&quot;Dispatches the request and on top of that performs request</pre>
<pre class="line after"><span class="ws">        </span>pre and postprocessing as well as HTTP exception catching and</pre>
<pre class="line after"><span class="ws">        </span>error handling.</pre></div>
</div>

<li><div class="frame" id="frame-140061202725904">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/stochss_compute/api/v1/gillespy2/model.py"</cite>,
      line <em class="line">70</em>,
      in <code class="function">run</code></h4>
  <div class="source "><pre class="line before"><span class="ws">    </span># Build a list of job key values which will need to be run.</pre>
<pre class="line before"><span class="ws">    </span>keys = [f&quot;{model_id}-trajectory_{i}&quot; for i in range(number_trajectories)]</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">    </span># Run each trajectory and save in a dataset.</pre>
<pre class="line before"><span class="ws">    </span># Warning: Here be possible threading issues. Needs investigation.</pre>
<pre class="line current"><span class="ws">    </span>dependencies = delegate.client.map(gillespy2.core.Model.run, [model] * number_trajectories, **run_request.kwargs, key=keys)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>def join_results(results):</pre>
<pre class="line after"><span class="ws">        </span># Save the dependent trajectories in memory.</pre>
<pre class="line after"><span class="ws">        </span>from distributed import get_client</pre>
<pre class="line after"><span class="ws">        </span>client = get_client()</pre></div>
</div>

<li><div class="frame" id="frame-140061202725520">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/werkzeug/local.py"</cite>,
      line <em class="line">432</em>,
      in <code class="function">__get__</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">                </span>return self.class_value</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">            </span>return self</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">        </span>try:</pre>
<pre class="line current"><span class="ws">            </span>obj = instance._get_current_object()</pre>
<pre class="line after"><span class="ws">        </span>except RuntimeError:</pre>
<pre class="line after"><span class="ws">            </span>if self.fallback is None:</pre>
<pre class="line after"><span class="ws">                </span>raise</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">            </span>return self.fallback.__get__(instance, owner)  # type: ignore</pre></div>
</div>

<li><div class="frame" id="frame-140061202724992">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/werkzeug/local.py"</cite>,
      line <em class="line">554</em>,
      in <code class="function">_get_current_object</code></h4>
  <div class="source library"><pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;Return the current object.  This is useful if you want the real</pre>
<pre class="line before"><span class="ws">        </span>object behind the proxy at a time for performance reasons or because</pre>
<pre class="line before"><span class="ws">        </span>you want to pass the object into a different context.</pre>
<pre class="line before"><span class="ws">        </span>&quot;&quot;&quot;</pre>
<pre class="line before"><span class="ws">        </span>if not hasattr(self.__local, &quot;__release_local__&quot;):  # type: ignore</pre>
<pre class="line current"><span class="ws">            </span>return self.__local()  # type: ignore</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span>try:</pre>
<pre class="line after"><span class="ws">            </span>return getattr(self.__local, self.__name)  # type: ignore</pre>
<pre class="line after"><span class="ws">        </span>except AttributeError:</pre>
<pre class="line after"><span class="ws">            </span>name = self.__name  # type: ignore</pre></div>
</div>

<li><div class="frame" id="frame-140061202765280">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/stochss_compute/api/v1/apiutils.py"</cite>,
      line <em class="line">18</em>,
      in <code class="function">get_delegate</code></h4>
  <div class="source "><pre class="line before"><span class="ws">    </span>delegate_type = current_app.config[&quot;DELEGATE_TYPE&quot;]</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line before"><span class="ws">    </span># # May be None</pre>
<pre class="line before"><span class="ws">    </span># kube_cluster = current_app.config[&quot;KUBE_CLUSTER&quot;]</pre>
<pre class="line before"><span class="ws"></span> </pre>
<pre class="line current"><span class="ws">    </span>delegate = delegate_type(delegate_config)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>if False in (delegate.connect(), delegate.test_connection()):</pre>
<pre class="line after"><span class="ws">        </span>raise Exception(&quot;Delegate connection failed.&quot;)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">    </span>return delegate</pre></div>
</div>

<li><div class="frame" id="frame-140061202764656">
  <h4>File <cite class="filename">"/home/brian/Desktop/research/stochss-compute/stochss_compute/api/delegate/dask_delegate.py"</cite>,
      line <em class="line">64</em>,
      in <code class="function">__init__</code></h4>
  <div class="source "><pre class="line before"><span class="ws">            </span>if self.delegate_config.kube_cluster is not None:</pre>
<pre class="line before"><span class="ws">                </span>self.client = Client(self.delegate_config.kube_cluster)</pre>
<pre class="line before"><span class="ws">                </span>print(self.delegate_config.kube_cluster)</pre>
<pre class="line before"><span class="ws">            </span># TODO what happens if user is not using kubernetes?</pre>
<pre class="line before"><span class="ws">            </span>else:</pre>
<pre class="line current"><span class="ws">                </span>self.client = Client(self.delegate_config.cluster_address)</pre>
<pre class="line after"><span class="ws"></span> </pre>
<pre class="line after"><span class="ws">        </span># Setup functions to be run on the schedule.</pre>
<pre class="line after"><span class="ws">        </span>def __scheduler_job_exists(dask_scheduler, job_id: str) -&gt; bool:</pre>
<pre class="line after"><span class="ws">            </span>return job_id in dask_scheduler.tasks</pre>
<pre class="line after"><span class="ws"></span> </pre></div>
</div>
</ul>
  <blockquote>AttributeError: &#x27;DaskDelegateConfig&#x27; object has no attribute &#x27;cluster_address&#x27;</blockquote>
</div>

<div class="plain">
    <p>
      This is the Copy/Paste friendly version of the traceback.
    </p>
    <textarea cols="50" rows="10" name="code" readonly>Traceback (most recent call last):
  File &quot;/home/brian/Desktop/research/stochss-compute/stochss_compute/api/delegate/dask_delegate.py&quot;, line 57, in __init__
    self.client = get_client()
  File &quot;/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/distributed/worker.py&quot;, line 4088, in get_client
    raise ValueError(&quot;No global client found and no address provided&quot;)
ValueError: No global client found and no address provided

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File &quot;/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py&quot;, line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File &quot;/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py&quot;, line 2076, in wsgi_app
    response = self.handle_exception(e)
  File &quot;/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py&quot;, line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File &quot;/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py&quot;, line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File &quot;/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py&quot;, line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File &quot;/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py&quot;, line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File &quot;/home/brian/Desktop/research/stochss-compute/stochss_compute/api/v1/gillespy2/model.py&quot;, line 70, in run
    dependencies = delegate.client.map(gillespy2.core.Model.run, [model] * number_trajectories, **run_request.kwargs, key=keys)
  File &quot;/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/werkzeug/local.py&quot;, line 432, in __get__
    obj = instance._get_current_object()
  File &quot;/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/werkzeug/local.py&quot;, line 554, in _get_current_object
    return self.__local()  # type: ignore
  File &quot;/home/brian/Desktop/research/stochss-compute/stochss_compute/api/v1/apiutils.py&quot;, line 18, in get_delegate
    delegate = delegate_type(delegate_config)
  File &quot;/home/brian/Desktop/research/stochss-compute/stochss_compute/api/delegate/dask_delegate.py&quot;, line 64, in __init__
    self.client = Client(self.delegate_config.cluster_address)
AttributeError: &#x27;DaskDelegateConfig&#x27; object has no attribute &#x27;cluster_address&#x27;</textarea>
</div>
<div class="explanation">
  The debugger caught an exception in your WSGI application.  You can now
  look at the traceback which led to the error.  <span class="nojavascript">
  If you enable JavaScript you can also use additional features such as code
  execution (if the evalex feature is enabled), automatic pasting of the
  exceptions and much more.</span>
</div>
      <div class="footer">
        Brought to you by <strong class="arthur">DON'T PANIC</strong>, your
        friendly Werkzeug powered traceback interpreter.
      </div>
    </div>

    <div class="pin-prompt">
      <div class="inner">
        <h3>Console Locked</h3>
        <p>
          The console is locked and needs to be unlocked by entering the PIN.
          You can find the PIN printed out on the standard output of your
          shell that runs the server.
        <form>
          <p>PIN:
            <input type=text name=pin size=14>
            <input type=submit name=btn value="Confirm Pin">
        </form>
      </div>
    </div>
  </body>
</html>

<!--

Traceback (most recent call last):
  File "/home/brian/Desktop/research/stochss-compute/stochss_compute/api/delegate/dask_delegate.py", line 57, in __init__
    self.client = get_client()
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/distributed/worker.py", line 4088, in get_client
    raise ValueError("No global client found and no address provided")
ValueError: No global client found and no address provided

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py", line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py", line 2076, in wsgi_app
    response = self.handle_exception(e)
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/brian/Desktop/research/stochss-compute/stochss_compute/api/v1/gillespy2/model.py", line 70, in run
    dependencies = delegate.client.map(gillespy2.core.Model.run, [model] * number_trajectories, **run_request.kwargs, key=keys)
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/werkzeug/local.py", line 432, in __get__
    obj = instance._get_current_object()
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/werkzeug/local.py", line 554, in _get_current_object
    return self.__local()  # type: ignore
  File "/home/brian/Desktop/research/stochss-compute/stochss_compute/api/v1/apiutils.py", line 18, in get_delegate
    delegate = delegate_type(delegate_config)
  File "/home/brian/Desktop/research/stochss-compute/stochss_compute/api/delegate/dask_delegate.py", line 64, in __init__
    self.client = Client(self.delegate_config.cluster_address)
AttributeError: 'DaskDelegateConfig' object has no attribute 'cluster_address'

-->
; pos=0; lineno=1; colno=1)

```

**Comment by briandrawert** [comment]
The terminal shows this error:
```
2021-11-20 10:50:12,028 - werkzeug - INFO - 127.0.0.1 - - [20/Nov/2021 10:50:12] "POST /api/v1/gillespy2/model/run HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/home/brian/Desktop/research/stochss-compute/stochss_compute/api/delegate/dask_delegate.py", line 57, in __init__
    self.client = get_client()
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/distributed/worker.py", line 4088, in get_client
    raise ValueError("No global client found and no address provided")
ValueError: No global client found and no address provided

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py", line 2091, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py", line 2076, in wsgi_app
    response = self.handle_exception(e)
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/brian/Desktop/research/stochss-compute/stochss_compute/api/v1/gillespy2/model.py", line 70, in run
    dependencies = delegate.client.map(gillespy2.core.Model.run, [model] * number_trajectories, **run_request.kwargs, key=keys)
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/werkzeug/local.py", line 432, in __get__
    obj = instance._get_current_object()
  File "/home/brian/Desktop/research/stochss-compute/venv/lib/python3.8/site-packages/werkzeug/local.py", line 554, in _get_current_object
    return self.__local()  # type: ignore
  File "/home/brian/Desktop/research/stochss-compute/stochss_compute/api/v1/apiutils.py", line 18, in get_delegate
    delegate = delegate_type(delegate_config)
  File "/home/brian/Desktop/research/stochss-compute/stochss_compute/api/delegate/dask_delegate.py", line 64, in __init__
    self.client = Client(self.delegate_config.cluster_address)
AttributeError: 'DaskDelegateConfig' object has no attribute 'cluster_address'
```

**Comment by mdip226** [comment]
this was fixed, but somehow it came back. basically, line 64 should just be `self.client = Client()`. but I just checked the file on the branch on github and it is correct. I'm not 100% sure but would that mean maybe your local branch is somehow not up to date?

**Comment by mdip226** [comment]
/home/brian/Desktop/research/stochss-compute/stochss_compute/api/delegate/dask_delegate.py

**Comment by mdip226** [comment]
it's commit 507de49

**Comment by ethangreen-dev** [comment]
One of these issues results from there being no Dask Cluster instance running at the port and IP defined in `DaskDelegateConfig`. Commit 507de49 fixed the issue because running `Client()` with no arguments automatically creates and connects to a local cluster @mdip226.

This actually sources from `self.client = Client(self.delegate_config.cluster_address)`, `cluster_address` is not a member of `DaskDelegateConfig`, it should actually be `self.client = Client(cluster_address)` where `cluster_address = $"{self.delegate_config.dask_cluster_address}:{self.delegate_config.dask_cluster_port}"`

I'll be issuing a fix with #48

**Comment by ethangreen-dev** [comment]
Fixed in PR #49
